tests/legacy-cli/e2e/tests/i18n/ivy-localize-serviceworker.ts

```diff
@@ -156,7 +158,7 @@ export default async function () {
     // Ivy i18n doesn't yet work with `ng serve` so we must use a separate server.
     const app = express();
     app.use(express.static(resolve(baseDir, lang)));
-    const server = app.listen(4200, 'localhost');
+    const server = app.listen(port, 'localhost');
```

Comment:
Oh this one should probably be updated to use the `externalServer` helper which starts the server. 

But your comment is still true within the `externalServer` method... is there a race condition though? I'm not really sure how port reserving(?) works. But we're using the `findFreePort()` method in many places so I'm not sure if switching this one place (`externalServer`) would make any difference?

It's also a bit awkward using `server.address()` because it returns `AddressInfo | string | null`, I'm not really sure when it would be a string, if we have to parse that string etc.

Reply:
The types are unfortunately awkward for this Node.js class.  When using `listen` with a port number, the address result will be an `AddressInfo` (or null if it has been closed or not yet listening).  It's only a string if it is listening on a pipe or unix domain socket. The code above actually has another issue in that listen is async but the callback is not used to ensure it is actually listening before proceeding with the test.

Reply:
I also notices the callback which was never used. I've updated to use the port selected by express and the callback which I think is working...